copyright.txt

```diff
@@ -0,0 +1,998 @@
+         These icons where heavily inspirted by https://github.com/elementary/icons
+
+Files: ui/pixmaps/com.github.xournalpp.xournalpp.svg
+       ui/pixmaps/com.github.xournalpp.xournalpp.png
+       ui/pixmaps/application-x-xojpp.svg
+       ui/pixmaps/application-x-xopp.svg
+       ui/pixmaps/application-x-xopt.svg
+       ui/pixmaps/gnome-mime-application-x-xopp.svg
+       ui/pixmaps/gnome-mime-application-x-xopt.svg
+       ui/pixmaps/xopt.svg
+Copyright: 2020, Luya Tshimbalanga <luya@fedoraproject.org>
+License: CC-BY-SA-4.0
+Comment: See https://github.com/xournalpp/xournalpp/issues/3103#issuecomment-844556592
+
+Files: cmake/find/FindFilesystem.cmake
+       cmake/include/clang-tidy.cmake
+Copyright: 2000-2021, Kitware, Inc. and Contributors
+License: BSD-3-clause
+
+Files: cmake/find/FindICU.cmake
+Copyright: 2011-2013, julp
+License: BSD-3-clause
+Comment: https://github.com/julp/FindICU.cmake
+         could maybe be replaced with
+         https://cmake.org/cmake/help/latest/module/FindICU.html
+
+Files: cmake/include/Copyright.txt
+Copyright: 2015, Marek Pikuła <marpirk@gmail.com>
+           Xournal++ team
+License: BSD-3-clause
+
+Files: cmake/include/ExternalDeb.cmake
+       cmake/include/GitRepo.cmake
+Copyright: 2015, Marek Pikuła <marek@pikula.co>
+License: BSD-3-clause
+
+Files: cmake/find/Gettext.cmake
+Copyright: 2012-2013, Valama development team
+License: GPL-3+
+
+Files: desktop/com.github.xournalpp.xournalpp.appdata.xml
+Copyright: 2019, Tobias Mueller <tobiasmue@gnome.org>
+License: CC0-1.0
+
+Files: src/util/hashcombine.h
+Copyright: 2019, Fabian Keßler <fabian_kessler@gmx.de>
+           1998-2021, David Eberly, Geometric Tools
+License: BSL-1
+
+Files: debian/docs
+       debian/package_description
+       debian/control
+       debian/changelog
+       debian/compat
+       debian/rules
+       debian/source/format
+Copyright: 2007, Mathieu Bouchard <mbouchar@bioinfo.ulaval.ca>
+           2008, Lucas Nussbaum <lucas@lucas-nussbaum.net>
+           2009-2014, Carlo Segre <segre@debian.org>
+           2018-2019, Andreas Butti <andreas.butti@gmail.com>
+           2020-2021, Barak A. Pearlmutter <bap@debian.org>
+License: GPL-2+
+
+Files: debian/copyright
+Copyright: 2018, Andreas Butti <andreasbutti@gmail.com>
+License: GPL-2+
+
+Files: plugins/MigrateFontSizes/plugin.ini
+Copyright: 2021, Roland Lötscher
+License: GPL-2+
+
+Files: plugins/ColorCycle/plugin.ini
+       plugins/LayerActions/plugin.ini
+Copyright: 2020, Roland Lötscher
+License: GPL-2+
+
+Files: plugins/Example/plugin.ini
+       plugins/ToggleGrid/plugin.ini
+Copyright: 2019, Andreas Butti
+License: GPL-2+
+
+Files: plugins/QuickScreenshot/plugin.ini
+Copyright: 2020, Debbie Reynolds
+License: GPL-2+
+
+Files: src/gui/GladeSearchpath.h
+Copyright: 2011, Andreas Butti
+License: GPL-2+
+
+Files: src/gui/dialog/LatexDialog.h
+Copyright: 2013, Wilson Brenna
+License: GPL-2+
```

Comment:
@rolandlo : To be honest I'm not sure. I noticed that in the xournalpp codebase no cpp file as a Licensing text (at least I didn't find any). So I assumed that this is somehow a convention in cpp programming to only put licensing/copyright on `.h` files. 
Do you have any experience with this?

Reply:
@idotobi I have no experience with this. In comparison, [Inkscape](https://gitlab.com/inkscape/inkscape/-/tree/master/src) has a copyright notice in every `.cpp`-file and every `.h`-file.

Reply:
There is definitely no such convention. Generally it is assumed that if there is no explicit copyright info in a file, the one of the repository root applies. So if it is not the same we must add the info to the file.

Reply:
Thanks for the confirmation @rolandlo and @LittleHuba I'll go through the files to check if we miss any headers.

For this particular one I'd actually be tempted to remove the licensing/copyright header as there's almost no resemblence between the last commit by Wilson Brenna and the current state of the file (see commit 0dc7b6e5e0e31804b1e4d65dcfe512614529270e).

But I guess as this is not that important since the license in anyway is the same as the one of the main project I'll just add the copyright/license information also to the `.cpp` file, just to be on the safe side.